src/nvim/lua/stdlib.c

```diff
@@ -549,19 +549,21 @@ static int nlua_iconv(lua_State *lstate)
 static int nlua_foldupdate(lua_State *lstate)
 {
   handle_T window = (handle_T)luaL_checkinteger(lstate, 1);
-  Error err = ERROR_INIT;
-  win_T *win = find_window_by_handle(window, &err);
-  if (ERROR_SET(&err)) {
-    nlua_push_errstr(lstate, err.msg);
-    api_clear_error(&err);
-    lua_error(lstate);
-    return 0;
+  win_T *win = handle_get_window(window);
+  if (!win) {
+    return luaL_error(lstate, "invalid window");
```

Comment:
if it's a false positive (i.e. "can never happen"), assert() or abort() is fine too

Reply:
`find_window_by_handle` in the old version uses `handle_get_window` and does null check internally, and sets `err` in that case. So `ERROR_SET(&err)` in the old version implies `win` is non-null. But adding assert to that is somewhat redundant, so I replaced those stuff to `handle_get_window` and direct null check.

Reply:
It's redundant either way, because that warning was a false positive.

Reply:
Right, but I just wanted to reduce the noise from the warning